### PR TITLE
Use specific hardhat version on openzeppelin and gp2 external tests

### PR DESCRIPTION
**Pull request by r0qs**
So we can enable those external tests in the CI again

### CODE REVIEW

File: test/externalTests/gp2.sh [hunk 1]
```diff
@@ -66,6 +66,14 @@ function gp2_test
...
     force_hardhat_unlimited_contract_size "$config_file" "$config_var"
     yarn
+    # We require to install hardhat 2.20.0 due to support for evm version cancun, otherwise we get the following error:
+    # Invalid value {"blockGasLimit":12500000,"hardfork":"cancun","allowUnlimitedContractSize":true} for HardhatConfig.networks.hardhat - Expected a value of type HardhatNetworkConfig.
+    # See: https://github.com/NomicFoundation/hardhat/issues/4176
+    yarn add hardhat@2.20.0
+
+    # Ignore bench directory which fails to compile with current hardhat and ethers versions.
+    # bench/trace/gas.ts:123:19 - error TS2339: Property 'equals' does not exist on type 'Uint8Array'.
+    jq '. + { "exclude": [ "bench" ] }' tsconfig.json > temp.json && mv temp.json tsconfig.json
```
Comment (cameel) [inline test/externalTests/gp2.sh]:
```suggestion
    jq '. + {"exclude": ["bench"]}' tsconfig.json > temp.json
    mv temp.json tsconfig.json
```

**Review by cameel — DISMISSED** [review]
Needs a small tweak but otherwise looks fine as long as it succeeds in CI.

**Review by cameel — APPROVED** [review]

**Comment by cameel** [comment]
I'd be careful with auto-merge here. It will merge the PR even if non-required jobs fail. And the two you're changing are not required :)

This should be merged only if it passes all the tests.

**Comment by nikola-matic** [comment]
Build is green across the board, so I'll be merging.

**Pull request merged**